frontend/packages/client/src/api/notificationService.js

```diff
@@ -0,0 +1,132 @@
+import {
+  LEANPLUM_APP_ID,
+  LEANPLUM_CRO_KEY,
+  LEANPLUM_EXPORT_KEY,
+} from 'api/constants';
+import Leanplum from 'leanplum-sdk';
+
+const COMMUNITY_UPDATES_CATEGORY_ID = 1;
+const options = {
+  method: 'GET',
+  headers: { accept: 'application/json' },
+};
+
+export const setUserId = async (walletId) => {
+  try {
+    Leanplum.setUserId(walletId);
+  } catch (e) {
+    throw new Error(e);
+  }
+};
+
+export const getUserSettings = async (walletId) => {
+  try {
+    fetch(
+      `https://api.leanplum.com/api?appId=${LEANPLUM_APP_ID}&clientKey=${LEANPLUM_EXPORT_KEY}&apiVersion=1.0.6&userId=${walletId}&action=exportUser`,
+      options
+    )
+      .then((response) => response.json())
+      .then((response) => {
+        const data = response.response[0];
+        if (!data.success) {
+          throw new Error(data.error.message);
+        }
+        if (!data.userAttributes) {
+          throw new Error('User Not Found');
+        }
+
+        const communitySubscriptions = [];
+        const res = {
+          email: data.userAttributes.email,
+        };
+        let isSubscribedToCommunityUpdates = true;
+
+        for (const property in data.userAttributes) {
+          if (property.includes('community')) {
+            const communityId = property.split('community')[1];
+            communitySubscriptions.push({
+              communityId,
+              subscribed: data.userAttributes[property] === 'True',
+            });
+          }
+        }
+
+        res.communitySubscription = communitySubscriptions;
+
+        if (data.unsubscribeCategories) {
+          data.unsubscribeCategories.forEach((category) => {
+            if (parseInt(category.id) === COMMUNITY_UPDATES_CATEGORY_ID) {
+              isSubscribedToCommunityUpdates = false;
+            }
+          });
+        }
+
+        res.isSubscribedToCommunityUpdates = isSubscribedToCommunityUpdates;
+        return res;
```

Comment:
The caller won't be able to access this returned value, checkout this example I put together: https://codepen.io/linxiaodapper/pen/poKozGv?editors=0012

Reply:
good catch here